### PR TITLE
Add jaeger support for frontend.

### PR DESCRIPTION
**Pull request by rghetia**

### CODE REVIEW

**Comment by rghetia** [comment]
@ahmetb PTAL.

File: src/frontend/main.go [hunk 1]
```diff
@@ -158,16 +182,26 @@ func initStats(log logrus.FieldLogger, exporter *stackdriver.Exporter) {
 }
 
 func initTracing(log logrus.FieldLogger) {
+	// This is a demo app with low QPS. trace.AlwaysSample() is used here
```
Comment (ahmetb) [inline src/frontend/main.go]: Hmm I recall we had this already in. I wonder at what point it got removed. Thanks for adding back.

File: src/frontend/main.go [hunk 2]
```diff
@@ -142,6 +143,29 @@ func main() {
 	log.Fatal(http.ListenAndServe(addr+":"+srvPort, handler))
 }
 
+func initJaegerTracing(log logrus.FieldLogger) {
+
+	svcAddr := os.Getenv("JAEGER_SERVICE_ADDR")
```
Comment (ahmetb) [inline src/frontend/main.go]: This env var doesn't match what we had in the Java adservice. It was a bool; here it's an addr with !="" check.

I think we should unify the discovery?

Comment (rghetia) [reply src/frontend/main.go]: yes, I plan to modify the jave to match this one. I like this approach because it allows the user to name the endpoints.

Comment (ahmetb) [reply src/frontend/main.go]: agreed this is better.

**Pull request merged**